### ISSUE TITLE
Copter: handle boat take-off and landings

### ISSUE THREAD

**Issue by rmackay9** [diydrones/ardupilot#1361]
This was requested as part of AC3.2 testing.
http://diydrones.com/forum/topics/arducopter-3-2-beta-testing?xg_source=msg_com_forum&id=705844%3ATopic%3A1656303&page=112#comments

There are two risks for arming and taking off on boats.  The first one is that the gyro calibration will likely fail.  It'll go with it's best guess which is likely ok most of the time but at least one case has been reported where the vehicle got a case of "the leans".  I.e. the pilot had to keep pulling further and further back on the pitch stick to keep it level... eventually there wasn't enough stick left and it flipped over.  This is because the gyro calibration didn't work very well.  Plane has a parameter that allows skipping the gyro cal, we don't have it in copter although it's easy to add.

The 2nd less risky issue is the pre-arm checks will fail if it thinks the vehicle is moving over 50m/s (i.e. 1.5feet / second).  This can be disabled by setting the ARMING_CHECK to Skip INS.

For Landing there may be an issue with the landing check's 30deg/sec rotation rate not being passed meaning the vehicle cannot be disarmed in AltHold or Loiter.

**Comment by govsux** [comment]
Is this going to be in the next Beta by summer?? Hope so. I'll be happy to test!

**Comment by Pascal-SoGeo** [comment]
I am also very interested in that function. I am looking toward offshore activity with UAV and obviously would need to take off from very large metallic and slow moving structures.

**Comment by rmackay9** [comment]
The arducopter part of the return-to-me is in master now and will go out with AC3.3.  This is one more step towards resolving this issue.

**Comment by TerryPts** [comment]
If the channel 9+ implementation is completed, one of the extra channels can be used to select "boat" mode which could set up the necessary parameters.

**Comment by rmackay9** [comment]
landings should be much better in AC3.3.  Take-offs (in particular the gyro calibration) could still fail and we will need to push that portion to AC3.4 'cuz we're simply out of time for AC3.3.

**Comment by nicksargeant** [comment]
This is disappointing but very much understandable given that boat operations are a pretty niche use case and there are plenty of other requests with wider applicability . Thanks for trying Randy! I am pretty keen to see this through, so if there is anything that can be done to help from an end-user perspective (testing/datalog collection) please let me know.

**Comment by CodeChief** [comment]
As a side note for when any development on this continues, you probably need something like GPS RTK to get enough accuracy to land on a boat deck. From what I understand so far 1cm accuracy is possible, but the base station must remain static so not sure what happens when the boat is moving.

Perhaps then it needs to just make the coordinates between the GPS on the boat and copter match rather than trying to get global accuracy. Actually if the boat has the gyro and compass too (potentially full APM hardware) then that should solve the calibration too?

Until a certain height is reached (and below when landing) we should match the horizonal of the boat anyway. When enough clearance is available (I guess 1m+ depending on size of drone) we can do some sort of gyro stabilization similar to the auto-tune stuff.

Just some random ideas, I find this fascinating. Logically we need need intra-APM communication mechanisms first, like Antenna Tracker but baked into the core. That could even bring new features like swarming, a good test case for that core feature. We need swarm/relative positioning to get the boat landing you see, it's kind of a variant of the same thing. The Antenna Tacker could also be refactored then to re-use the same code.

**Comment by nicksargeant** [comment]
Precision landing is largely a separate issue and there are numerous solutions being developed  (see ir-lock for example). A competent pilot should be able to land a multirotor on deck, but before you worry about landing, the multirotor needs to be able to initialise, arm and deploy from a boat.

**Comment by rmackay9** [comment]
@CodeChief, the idea to use a separate APM for the gyro cal (or to speed up the EKF stabilizing) is an interesting one.  I'll pass it onto Paul Riseborough.

**Comment by rrr6399** [comment]
I ran into this issue recently as well. We could not get the copter to pass the prearm checks (velocity error and gyro health errors.)  I started an experiment to disable the prearm checks, but ran out of time. 

I was wondering if it'd be possible to perform the prearm checks on land before getting on the boat and having any required calibration data saved. Then at sea, the saved data would be used (except for home) for calibration and safety check purposes. Even though this process is not ideal, would it work?

**Comment by rmackay9** [comment]
@rrr6399, yes if you arm once on the ground and then put the copter into the boat then it should be ok.

**Comment by govsux** [comment]
Arming on the ground then getting on the boat isn't real practical at least for me because where I fly is a mile from the boat ramp. About the only way I was able to have it work was to hold the quad and do my very best to keep it steady while the boat was rocking with the waves as it armed. It doesn't work very well. Hanging it by a string/s would work but again not real practical. Seems like if there was a way to have ch7-9 allow 'boat mode' which you would arm the copter on land then 'save' those calibrations to 'boat mode calibrations' then the next time you fly in boat mode it would use those.

**Comment by govsux** [comment]
re-  This can be disabled by setting the ARMING_CHECK to Skip INS.
Would this effectually then use the last know calibrations?

**Comment by rmackay9** [comment]
Yeah, the arming on land isn't a long-term solution.  I talked with Paul Riseborough this morning and I'm pretty certain we will get the improved gyro-cal which doesn't require the copter to be completely stable into AC3.4.
Disabling the arming-checks helps some but there's still one gyro calibration on the first arming that isn't optional.  It'll try for up to 30seconds but if it can't get a good calibration it'll fail to arm.

**Comment by rrr6399** [comment]
Why does gyro calibration have to be done every power on? (Or arm?) Could it use last known good instead as a workaround? That would allow cals on level land, power down and then power up the boat, arm and launch.

Or, does it currently need a stable state at takeoff to reference?

**Comment by R-Lefebvre** [comment]
Actually I thought we did have an option to allow booting without calibrating gyro. It uses the numbers from parameters. I think Triage does this  with planes?

**Comment by rmackay9** [comment]
the SKIP_GYRO_CAL param.. indeed.  sadly it's only implemented in Plane.  I looked at adding it to copter but I saw too much drift.  If it was implemented, I imagine it would work, you'd just need to leave the copter a long time to let it learn the change in offsets from the previous bot.

**Comment by govsux** [comment]
Dang. Need a sensor that only see gravity? I work in construction and use a rotating laser level. It finds level right away. Then again it has an advantage it's spinning. And I never used it in a boat :] Would there be a way of asking the gyro to take snap shots of the full 360 [ with the help of the compass ] simulating spinning then average it out? Or maybe that's what it basically already does? Easy for me to say I'm just the carpenter.

**Comment by rmackay9** [comment]
@govsux, it just measures it's rotation rate really quickly in very short bursts and then averages the results.  It does that a couple of times and checks that all the averages match.  The problem that they don't match if the boat is rotating left, then rotating right relatively slowly.
What we need to do is extend that period of averaging so that it's really long.  So say the boat rocks back and forth maybe once every 5 seconds.. we need to average over a 10s period.

**Comment by govsux** [comment]
Just fyi the time frame back and forth is more like 2-4 seconds. Splitting hairs.. but just saying, Because 5 seconds plus probably starts to get to boats much bigger and thus not even having a problem. 
If the options were A- start by arming whilst bending your knees then double check it worked by holding the copter and going 1/2 throttle to check if it tries to stay level or off by a few degrees [ while the boat is rocking no less ] OR B- arm copter like normal but wait 30-60 seconds I'd got with B every time! If that was the beginning of 'beta boat mode'  I'd say that's a great start!

**Comment by rrr6399** [comment]
Randy, what is the reason for the gyro drift? When you say leave it for long time, do you mean let it sit on the ground after powering it up for a long time or arming it and leaving it for a long time or something else

**Comment by rmackay9** [comment]
@rrr6399, The gyro offsets seem to change quite a bit from boot to boot.  Maybe it's the temperature change or maybe it's just that we need the offsets to be really accurate.  In any case, if we don't get those offsets really close (i.e. within 0.1 deg/sec) then you'll see the HUD rotate badly especially after arming because at that moment we increase reliance on the gyros.
The EKF will learn the gyro offsets at the rate of about 1 deg/sec.  So if you manage to arm it once and then leave it sitting on the ground (or even leave it on the deck of a moving boat) long enough it'll get the offsets right and the HUD will remain stable.

**Comment by govsux** [comment]
Any movement on this?

**Comment by rmackay9** [comment]
@govsux, not yet - stlil working on getting AC3.3 out the door, but AC3.4 will certainly include this ability.

**Comment by govsux** [comment]
@rmackay9 Good to know. Thank you! Can't wait to test it out. Still flying 3.2 [ seems like you have a lot of testers for 3.3  so I passed this phase ] but will be more than happy to test 3.4 when it's ready :]

**Comment by antslake** [comment]
I recently did a "job" filming boat races from a boat. We had a lot of difficulty arming. I definitely could user a boat mode. Speaking from ignorance, would an alternate way of doing this would be to have a second gyro that you initialize on shore, then somehow user that as a reference for the main gyro. A virtual gimbal so to speak? Second problem is initializing gimbals.

**Comment by ballan33** [comment]
Could you arm on land and then just keep it powered? Obviously you would need to change batteries, but could this be done by hot swapping on a parallel adapter? I understand this isn't ideal, but would it work for the time being? As long as the copter doesn't turn off, you don't need to do a re-calibration.

I'm still fairly new to multirotors, so please let me know if there is a reason this would be a bad idea.

Thanks

**Comment by antslake** [comment]
Last boat race I filmed was 10 hours long. Of course you aren't filming the whole time, but it is impractical to keep it powered that long. I also had no access to the shore until tthe race was done, so if something happened and I lost power, I would be out of luck.

**Comment by magicrub** [comment]
There is a large ekf update in the pipeline. One of its new features will
be arming while on a boat. No ETA, just giving you a little hope.

**Comment by nicksargeant** [comment]
@magicrub, are you refering to Paul Riseborough's post here: https://groups.google.com/d/msg/drones-discuss/fP6l7EnOF88/aEL1Z9eqq0YJ
Very excited for this.

**Comment by magicrub** [comment]
Yup

**Comment by ballan33** [comment]
Thanks @antslake. I know it would be difficult, but keeping a single battery (8000 4S in my case) just for powering the multirotor on the boat but not for flying would last all day. This would mean you only really need to carry 1 more battery than you would have to normally.

I understand that 1 mistake with battery changeover would render the multirotor unusable, but my bigger question is whether it would be safe to hot swap using a parallel adapter? After a flight you'd be going from a battery at 20-30% capacity and connecting a battery at 100% capacity. I know they will start to balance between themselves, but unplugging one and plugging in another would only take seconds. Does anyone know if the abrupt change in voltage would damage the batteries/multirotor?

Thanks

**Comment by govsux** [comment]
re- battery swap No that wouldn't hurt anything I don't think.
Lets keep our fingers crossed on that EKF update!

**Comment by AndKe** [comment]
@ballan33 depending on internal resistance of your batteries, connecting a full and depleated pack will cause high current. A diode and lower voltage ground power would fix that.
Voltage changes are not a problem

**Comment by ballan33** [comment]
Thanks @AndKe, I was thinking of using XT-90S anti-spark connectors to help with sparking issues, but I was wondering if you could provide more information on the type of diode set-up you describe?

Thanks!

**Comment by govsux** [comment]
@ballan33 I connected one fully charged pack with one drained pack in parallel for 15 seconds without issue. Small gauge wire between them didn't even get warm. Very small spark similar to what you see just connecting a pixhawk. So the internal resistance of the pack is minimal. Not much current flow between the battries. I think you'd be fine with just a basic Y connection.

**Comment by ballan33** [comment]
Thanks very much for testing it @govsux, and fantastic result! This gives me an option for my  splashdrone (modified with APM) until a better solution comes along.

**Comment by R-Lefebvre** [comment]
Another option would be to use a special harness based on female balance plugs, using only the positive and negative wires, and some resistors to reduce the power flow between batteries.  When doing the battery hand-off, you'd first connect the special balance harness to the old battery, disconnect the main connector from the old battery.  Then connect the balance harness to the new battery. Disconnect the balance harness from the old battery, then connect the main connector to the new battery.

Slightly complicated and overkill, but it's a thought.

**Comment by govsux** [comment]
Not to be  pest just hoping there is some movement here. I flew off my boat this weekend I haven't done so since a couple versions ago pre 3.2  I think,,,, Something changed I could not get it to arm at all unless I skipped the pre arm checks and even then it took a LONG time to calibrate and arm. After I took off it sort of wigged and said it had a EKF variance.. scary as I just installed my first gopro and I was of course over water. All ended well... but I would sure love not turing off checks. Hoping boat mode get's the nod for 3.4. I wish I could do more than just complain but code isn't my cup of tea. But as mentioned I would be happy to test and knowingly take the risk over water for the cause. Without the gopro though :]

**Comment by CodeChief** [comment]
+1 for that, but perhaps it should be called/developed as a more generic "Take-off from moving platform" mode (with a shorter name of course) rather than the specific "boat mode". I'm thinking about those FliteTest videos where they launch a plane from a moving quad bike, an "aircraft carrier" or "roving home" in a mission to Mars scenario. All professional examples that justify some development.

I think base stations will become more attractive (think RTK 1cm GPS accuracy, WiFi relay and antenna tracker features) as these various libraries/components are updated to support them. Professionally they are really a requirement.

Taken further, multiple "stations" or "mobile platforms" could be defined. Think drone delivery or RTK reference points for stationary "stations". I was thinking about mesh networking and the like. But that's way off in the future. To start with we just need to take-off relative to the platform we are sitting on.

Perhaps "Platform" mode is a nice short name for this?

**Comment by rmackay9** [comment]
@priseborough and I discussed this again a few days ago.  There are a few changes required but the most important is the removal of the 2nd gyro calibration that happens when you arm (just the first time you arm).  We need to get rid of that and we know basically how to do that so it's just a matter of time.  It'll be in AC3.4 for sure though.

**Comment by priseborough** [comment]
My work on the updated EKF maths is sitting here:

https://github.com/priseborough/ardupilot/tree/ekfNewMaths-wip

So far it has only flown on an Iris, but did demonstrate its ability self align whilst moving. It needs to be rebased again on Master (basically replace the entire AP_NavEKF.cpp and AP_NavEKF.h file).

Takeoffs from a steel deck with large magnetic interference could be an issue unless we delay using GPS for the first few metres of height gained or use some other external data to initialise heading.

I plan to resume work on this branch in a few weeks.

-Paul

**Comment by govsux** [comment]
re-Takeoffs from a steel deck with large magnetic interference could be an issue unless we delay using GPS for the first few meters of height gained or use some other external data to initialize heading.

%99 of users aren't on a steel deck I don't think?? But either way thank you for working on this!! No pressure but summer [ boating weather ] is winding down for us poor chaps in the norther hemisphere. Thanks

**Comment by rmackay9** [comment]
Copter-3.3 includes the metal deck fixes from Paul including a reset of the compass heading once the vehicle is about 1.5m off the deck.  It's more the compass that gets upset rather than the GPS I think.

**Comment by priseborough** [comment]
We can  take off using GPS on a metal deck, if we are prepared to trust our compass readings to come good when the height increases. it just means we can't check compass health before takeoff unless there is another compass co-located that sees the same field.

**Comment by rrr6399** [comment]
FYI: I noticed that the Mikrokopter folks handled the boat launches by saving the gyro calibration parameters during accelerometer calibration and then allowed the pilot to optionally choose to use the saved gyro settings when taking off from a moving platform:
http://wiki.mikrokopter.de/en/Firmware-2.08#Boat_Mode:_Switch_on.2C_calibrate_and_start_under_moving_conditions

**Comment by rmackay9** [comment]
The ability to disable the gyro calibration is in master now and will go out with Copter-3.4.
The parameter to control the behaviour is INS_GYR_CAL:
0 = never do gyro calibration
1 = do once at startup
2 = do once at startup and once again during first arming

"1" is the new default for Copter.  For "boat mode" the user should set this param to "0".

**Comment by magicrub** [comment]
I know this is probably too late to mention, but why use _GYO_ instead of
_GYRO_ ?

**Comment by rmackay9** [comment]
@magicrub, the other gyro related parameters (like INS_GYR_OFFSET) all used just "_GYR_" instead of "_GYRO_".

**Comment by govsux** [comment]
Great!!!!!! Thanks!!! Any idea when 3.4 beta will be available?

**Comment by rmackay9** [comment]
@govsux, it'll be months and months before it's the official release but we will probably do some beta releases in a couple of months.  Summer is slipping away in the northern hemisphere so I know that's not ideal but..

**Comment by govsux** [comment]
Thanks Randy. I'm just glad there seems to be a solution. Guess I'll just have to go to FL this winter to test it out. whooooo is me.. ha

**Comment by ballan33** [comment]
We're coming into summer in Australia (the land that has forgotten snow), and I'm happy to test it Randy.

**Comment by rrr6399** [comment]
FYI: I ran a simple test in the park where I performed simple hover test in guided and auto mode using INS_GYR_OFFSET=1 with prearm checks with the copter at rest. Then I set INS_GYR_OFFSET=0, cycled power and moved the copter around continuously to attempt to simulate a rocking boat until I got GPS lock. Then performed the hover test in guided and auto mode again. It worked flawlessly for this test! Looks promising!

**Comment by rmackay9** [comment]
@rrr6399, cool, thanks for testing!

**Comment by govsux** [comment]
I thought this wasn't available until beta 3.4?? How long has that param setting been around?

**Comment by rrr6399** [comment]
I used the latest 3.4 development version for testing despite the risks:
http://firmware.diydrones.com/Copter/latest/

(along with the beta version of the Mission Planner.)

**Comment by govsux** [comment]
oh I see... thanks. Only thing is I don't know how to direct APM Planner [ Mac user ] to use the downloaded file. I'll have to look into that. Obviously I'm not a 'code' guy :]

**Comment by rrr6399** [comment]
Yeah, it looks like APM Planner doesn't support loading downloaded firmware like Mission Planner does. From this page http://firmware.diydrones.com/, there is a section on uploading firmware on Linux and Mac:

To load a PX4FMU firmware on a Linux or MacOS machine you will need to use the px_uploader.py python script. You can run it like this:
python px_uploader.py --port /dev/ttyACM0 px4fmu.px4
After starting the script, press the reset button on your PX4FMU to make it enter bootloader mode.

Be sure to recalibrate your copter after the firmware update.

**Comment by govsux** [comment]
great many thanks rrr6399!

**Comment by billbonney** [comment]
You can upload custom FW in APM Planner 2.0, you just need to file you want to install downloaded to your HD. You will need to select File->Advanced Mode to see the option in the Install Firmware view.
![firmwarecustomintsall](https://cloud.githubusercontent.com/assets/959994/10682361/b193199e-78e9-11e5-8134-a6c870e5442c.png)

**Comment by govsux** [comment]
@ Bill ohhh gotcha so the 'custom' button... I should have figured that out. Thanks!

**Comment by govsux** [comment]
Make sure you click on the 'advanced' button under the file menu. That might be it.

**Comment by rrr6399** [comment]
You're right! Who would've known! :-)

**Comment by govsux** [comment]
FYI I installed the dev version PX4v2 0036002D 30345101 37383933 and tested 'boat mode' and it worked great!! My boat is put away for the winter but I armed it holding it in my hand [ which was almost impossible with 3.3 ] and it worked flawlessly. Several times I armed it whilst moving the copter around as much or more than I would ever expect in my boat and each time it flew perfectly level just as though I had taking off from a solid surface. I did not do anything with the params, settings, didn't even recalibrate the compass or IMU's it just worked great right out of the box so to speak. I did not do any other testing of the code just as mentioned above. It does seem to take a few seconds longer to boot up the pixhawk than before. But no error codes thrown. Seems good :]

**Comment by rmackay9** [comment]
@govsux, excellent, thanks for testing!

**Comment by TerryPts** [comment]
This looks very promising. Thanks Rick for your tests and Randy for your hard work. Any chance that 3.4 will be released before spring?

**Comment by rmackay9** [comment]
@TerryPts, yes, very likely before spring.  We're working with the Solo guys to merge their fork with master and the most efficient thing to do would be to start testing Copter-3.4 in early Jan.  That will mean Copter-3.4 will be a little lighter on features than originally planned but this one would be included for sure because it's already done.

**Comment by govsux** [comment]
I'd like to test 'boat mode' in the FL keys over the Christmas holidays while I'm there. Not to mention some great video.  Have you been flying dev 3.4 for long Randy? I know there's no guarantees but is there anything with the code that seems iffy? I'll pass if there are for sure known bugs not ironed out. Either way it's at my own risk of course.

**Comment by rmackay9** [comment]
@govsux, I fly master quite often and generally it's fine.
It changes nearly every day though so it can be fine one day and not the next.  We always try to keep master reliable and flyable but sometimes we make a mistake.  Besides that, it's difficult for the dev team to support people flying master because we can't make assumptions about what's in/out of the code so we have to consider the possibility of bugs in all parts of the code.
Anyway, in short, my guess is you'll be fine but you know.. as you say, it's at your own risk.

**Comment by govsux** [comment]
Yea that all makes sense thanks Randy

**Comment by govsux** [comment]
FYI 3.4  PX4v2 0036002D 30345101 37383933   auto, pos hld, alt hld, stab, no noticeable problems. It's nice not to have to baby sit the thing for 30 seconds until it's done booting. :]  Seems like it's not as good in loiter / pos hold as 3.3 but hard to tell probably just me. I set boat mode to '3' aka never re-cal imu's. . Many flights no issue arming even whilst moving.

**Comment by govsux** [comment]
FYI 3.4 PX4v2 0036002D 30345101 37383933  I had many many flights launching and catching by hand in my boat over a 2-3 week period  with no issues. Awesome. Thanks!

**Comment by nicksargeant** [comment]
Thank you for the report @govsux. I look forward to beta testing "boat mode" in 3.4!

**Comment by StepperUser** [comment]
@govsux , I've tried to put FW beta v3.4 but there is no "boat mode" neither something called alike, I've been looking each parameter and lot of them have been renamed in comparision with v3.3 and have no description. How did you did it? I'm very interested in testing this "boat mode" and don't wanna wait until 3.4 stable release :)

**Comment by rmackay9** [comment]
@StepperUser, there's no new "boat mode" so to speak.  Basically you can just power-on and take-off in regular Loiter mode (or whatever flight mode you want) and it should just work.

**Comment by StepperUser** [comment]
Yes, I tested yesterday and it works, thank you @rmackay9! :100: 
It's a pitty that in order to know which parameter is which because some of them have been renamed, but for sure will be solved for the stable release :)

**Comment by StepperUser** [comment]
@rmackay9, I think I found a bug. When powering the quadricopter, if I try to arm before finding 9 satellites then it will never arm, have to disconnect, connect, wait until reaches the number of satellites and then it works perfectly. Amazing work what you are doing with v3.4!

**Comment by OXINARF** [comment]
@StepperUser that's a support question, please post in the forum (http://discuss.ardupilot.org) together with a log.

**Comment by StepperUser** [comment]
Done, @OXINARF , sorry for my mistake.

**Comment by antslake** [comment]
Now that we can arm our copters on a rocking boat, has anyone figured out how to arm the gimble for the camera?

**Comment by lvale** [comment]
@antslake post in the forum (http://discuss.ardupilot.org)

**Comment by rmackay9** [comment]
@StepperUser, The issue will most likely be that the circular fence is enabled.  To enforce the fence it needs to know it's position so it requires the GPS..

**Comment by ytlee** [comment]
For boat landing, can we program a "kill" switch that instantly cut off all the motors, as for boat takeoff, can the user do the gyro on land first, when it is in boat mode, just skip the calibration and use existing (previous) value.

**Comment by lvale** [comment]
@ytlee This might not be the best place for support questions, but yes, just program one of the options as the kill switch.

**Comment by govsux** [comment]
@ytlee I have thought the same thing. I hand catch mine now because the odds of me bumping something with a prop is very high trying to land in such a tight space. I have often thought if I had a small net suspended in my boat that if I could kill the motors whilst I have the quad directly above the net it would catch it nicely.

**Comment by lvale** [comment]
https://www.youtube.com/watch?v=-Db4u8LJE5w

http://ardupilot.org/copter/docs/channel-7-and-8-options.html?highlight=ch7#auxiliary-function-switches

**Comment by R-Lefebvre** [comment]
Wow, those are some fantastic features. :)

**Comment by dposea** [comment]
I thought I would toss in my $1000 testing with a Phantom. 23 foot Grady White in 3-4 foot seas. The only really decent launch spot was the top of the transom where there is a 15"x15" cooler built in. After arming the drone would speed up and slow down props to compensate for wave action. To launch I just punched the throttle, rocketing up 100 feet. No issues.

Landing is almost impossible. You can not match the deck lean to the drone without the drone moving in the lean direction. On about attempt #4 I hit an outrigger and drone went to visit the fishes :(

Second attempt, all rods not in holders, riggers extended outwards, brave buddy in the back of the boat. Speed increased to about 10 mph to steady out the boat, and direction picked to minimize roll. I just flew the drone right over the motor then slowly guided it into grab range. He grabbed the gear and I instantly killed the rotors.
